app/src/main/java/com/keylesspalace/tusky/util/twittertext/Regex.java

```diff
@@ -0,0 +1,348 @@
+// Copyright 2018 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Tusky changed: slight adaptions for Mastodon compatibility
+
+package com.keylesspalace.tusky.util.twittertext;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.regex.Pattern;
+import javax.annotation.Nonnull;
+
+public class Regex {
+
+  protected Regex() {
+  }
+
+  private static final String URL_VALID_GTLD =
+    "(?:(?:" +
+    join(TldLists.GTLDS) +
+    ")(?=[^a-z0-9@+-]|$))";
+  private static final String URL_VALID_CCTLD =
+    "(?:(?:" +
+    join(TldLists.CTLDS) +
+    ")(?=[^a-z0-9@+-]|$))";
```

Comment:
Kind of annoying to have these static TLD lists that we will have to keep updating 😐

Reply:
yes, but without them we could end up with a similar problem as #4425 when someone posts https://nonexisting.domain.whatever :/